File: Plugins/Aspose_Tasks_Java_for_Ruby/lib/asposetasksjava/Tasks/tasksproperties.rb
module Asposetasksjava
  module TasksProperties
    def initialize()
        # Getting General Properties
        get_general_properties()

        # Getting Actual Properties
        get_actual_properties()

        # Getting Outline Properties
        get_outline_properties()
    end

    def get_general_properties()
        data_dir = File.dirname(File.dirname(File.dirname(File.dirname(__FILE__)))) + '/data/'
                
        # Instantiate project object
        project = Rjb::import('com.aspose.tasks.Project').new(data_dir + 'test_tasks.mpp')

        # Create a ChildTasksCollector instance
		collector = Rjb::import('com.aspose.tasks.ChildTasksCollector').new

		# Collect all the tasks from RootTask using TaskUtils
		Rjb::import('com.aspose.tasks.TaskUtils').apply(project.getRootTask(), collector, 0)

		tasks = collector.getTasks()
		#puts tasks.size()
		#abort()

		tsk = Rjb::import('com.aspose.tasks.Tsk')

		# Parse through all the collected tasks
		#tasks.each do |task|
		i = 0
		while i < tasks.size()
			task = tasks.get(i)
		    puts "Task Id:" + task.get(tsk.ID).to_string
		    puts "Task Uid: " + task.get(tsk.UID).to_string
		    puts "Task Name: " + task.get(tsk.NAME).to_string
		    puts "Task Start: " + task.get(tsk.START).to_string
		    puts "Task Finish: "+  task.get(tsk.FINISH).to_string
		    puts "---------------------------------------------"
		    i +=1
		end
    end

    def get_actual_properties()
        data_dir = File.dirname(File.dirname(File.dirname(File.dirname(__FILE__)))) + '/data/'
                
        # Instantiate project object
        project = Rjb::import('com.aspose.tasks.Project').new(data_dir + 'test_tasks.mpp')

        # Create a ChildTasksCollector instance
		collector = Rjb::import('com.aspose.tasks.ChildTasksCollector').new

		# Collect all the tasks from RootTask using TaskUtils
		Rjb::import('com.aspose.tasks.TaskUtils').apply(project.getRootTask(), collector, 0)

		tasks = collector.getTasks()

		tsk = Rjb::import('com.aspose.tasks.Tsk')

		# Parse through all the collected tasks
		i = 0
		while i < tasks.size()
			task = tasks.get(i)
		    puts "Task Name : " + task.get(tsk.NAME).to_string
			puts "Actual Start: " + task.get(tsk.ACTUAL_START).toString()
			puts "Actual Finish: " + task.get(tsk.ACTUAL_FINISH).toString()
			puts "Actual Duration: " + task.get(tsk.ACTUAL_DURATION).toString()
			puts "Actual Cost: " + task.get(tsk.ACTUAL_COST).toString()
			puts "---------------------------------------------"
		    i +=1
		end
    end

    def get_outline_properties()
        data_dir = File.dirname(File.dirname(File.dirname(File.dirname(__FILE__)))) + '/data/'
                
        # Instantiate project object
        project = Rjb::import('com.aspose.tasks.Project').new(data_dir + 'test_tasks.mpp')

        # Create a ChildTasksCollector instance
		collector = Rjb::import('com.aspose.tasks.ChildTasksCollector').new

		# Collect all the tasks from RootTask using TaskUtils
		Rjb::import('com.aspose.tasks.TaskUtils').apply(project.getRootTask(), collector, 0)

		tasks = collector.getTasks()

		tsk = Rjb::import('com.aspose.tasks.Tsk')

		# Parse through all the collected tasks
		i = 0
		while i < tasks.size()
			task = tasks.get(i)
		    puts "Outline Level: " + task.get(tsk.OUTLINE_LEVEL).to_string
			puts "Outline Number: " + task.get(tsk.OUTLINE_NUMBER).to_string
			puts "---------------------------------------------"
		    i +=1
		end
    end
  end
end

File: Plugins/Aspose_Tasks_Java_for_PHP/src/aspose/tasks/WorkingWithTasks/TasksProperties.php
<?php
namespace Aspose\Tasks\WorkingWithTasks;

use com\aspose\tasks\Project as Project;
use com\aspose\tasks\ChildTasksCollector as ChildTasksCollector;
use com\aspose\tasks\TaskUtils as TaskUtils;
use com\aspose\tasks\Tsk as Tsk;
class TasksProperties{

    public static function run($dataDir=null){

        # Getting General Properties
        TasksProperties::get_general_properties($dataDir);

        # Getting Actual Properties
        TasksProperties::get_actual_properties($dataDir);

        # Getting Outline Properties
        TasksProperties::get_outline_properties($dataDir);
    }

    public static function get_general_properties($dataDir=null){

        # Instantiate project object
        $project = new Project($dataDir . 'test_tasks.mpp');

        # Create a ChildTasksCollector instance
		$collector = new ChildTasksCollector();

        # Collect all the tasks from RootTask using TaskUtils
        $taskUtils=new TaskUtils();
        $taskUtils->apply($project->getRootTask(), $collector, 0);
		$tasks = $collector->getTasks();
		#puts tasks.size()
		#abort()

        $tsk = new Tsk();

		# Parse through all the collected tasks
		#tasks.each do |task|
		$i = 0;
		while ($i < sizeof($tasks)) {
            $task = $tasks -> get($i);
		    print "Task Id:" . $task -> get($tsk -> ID).PHP_EOL;// . to_string
		    print "Task Uid: " . $task -> get($tsk -> UID).PHP_EOL;// . to_string
		    print "Task Name: " . $task -> get($tsk -> NAME).PHP_EOL;// . to_string
		    print "Task Start: " . $task -> get($tsk -> START).PHP_EOL;// . to_string
		    print "Task Finish: " . $task -> get($tsk -> FINISH).PHP_EOL;// . to_string
		    print "---------------------------------------------".PHP_EOL;
		    $i += 1;
		}
    }

    public static function get_actual_properties($dataDir=null){

        # Instantiate project object
        $project = new Project($dataDir . 'test_tasks.mpp');

        # Create a ChildTasksCollector instance
		$collector = new ChildTasksCollector();

            # Collect all the tasks from RootTask using TaskUtils
        $taskUtils=new TaskUtils();
        $taskUtils->apply($project->getRootTask(), $collector, 0);

		$tasks = $collector->getTasks();

		$tsk = new Tsk();

		# Parse through all the collected tasks
		$i = 0;
		while ($i < sizeof($tasks)) {
            $task = $tasks -> get($i);
		    print "Task Name : " . (string)$task -> get($tsk -> NAME).PHP_EOL;// . to_string
			print "Actual Start: " . (string)$task -> get($tsk -> ACTUAL_START).PHP_EOL;// . toString()
			print "Actual Finish: " . (string)$task -> get($tsk -> ACTUAL_FINISH).PHP_EOL;// . toString()
			print "Actual Duration: " . (string)$task -> get($tsk -> ACTUAL_DURATION).PHP_EOL;// . toString()
			print "Actual Cost: " . (string)$task->get($tsk -> ACTUAL_COST).PHP_EOL;// . toString()
			print "---------------------------------------------".PHP_EOL;
		    $i += 1;
		}

    }

    public static function get_outline_properties($dataDir=null){

        # Instantiate project object
        $project = new Project($dataDir . 'test_tasks.mpp');

        # Create a ChildTasksCollector instance
		$collector = new ChildTasksCollector();

            # Collect all the tasks from RootTask using TaskUtils
        $taskUtils=new TaskUtils();
        $taskUtils->apply($project->getRootTask(), $collector, 0);

		$tasks = $collector->getTasks();

		$tsk = new Tsk();

		# Parse through all the collected tasks
		$i = 0;
		while ($i<sizeof($tasks)) {
            $task=$tasks->get($i);
		    print "Outline Level: " . (string)$task->get($tsk -> OUTLINE_LEVEL).PHP_EOL;// . to_string;
			print "Outline Number: " . (string)$task->get($tsk -> OUTLINE_NUMBER).PHP_EOL;// . to_string;
			print "---------------------------------------------".PHP_EOL;
		    $i += 1;
		}

    }

}
?>